source/dub/recipe/json.d

```diff
@@ -54,13 +55,16 @@ void parseJson(ref PackageRecipe recipe, Json json, string parent_name)
 	recipe.buildSettings.parseJson(json, fullname);
 
 	if (auto pv = "configurations" in json) {
-		TargetType deftargettp = TargetType.library;
-		if (recipe.buildSettings.targetType != TargetType.autodetect)
-			deftargettp = recipe.buildSettings.targetType;
-
 		foreach (settings; *pv) {
 			ConfigurationInfo ci;
-			ci.parseJson(settings, recipe.name, deftargettp);
+			ci.parseJson(settings, recipe.name);
+
+			if (recipe.buildSettings.targetType == TargetType.autodetect &&
+				ci.buildSettings.targetType == TargetType.autodetect) {
+				logWarn("Warning: Configuration '%s' has no targetType, using 'library' as default.", ci.name);
```

Comment:
This warning should be output in `Package.simpleLint` instead, because otherwise every build will be spammed with warnings if any available package on the system runs into this case, no matter if used or not.